docs/tutorial/efficient-dataflow.md

```diff
@@ -157,19 +168,19 @@ Then we add necessary transformations:
 .. code-block:: python
     :emphasize-lines: 3-5
 
-    ds = LMDBData(db, shuffle=False)
+    ds = LMDBData(db, shuffle=False) # produces key-value pairs
     ds = LocallyShuffleData(ds, 50000)
-    ds = LMDBDataPoint(ds)
-    ds = MapDataComponent(ds, lambda x: cv2.imdecode(x, cv2.IMREAD_COLOR), 0)
+    ds = LMDBDataPoint(ds)  # converts 'ds' to  pairs (encoded_image, label)
+    ds = ImageDecode(ds, mode='.jpg', index=0)
```

Comment:
Also, if it is provided as a dataflow called `ImageDecode`, people would want to use it with one datatype or another (at least bytes and nparray are common). Similarly for encode, we cannot assume users want an nparray in the end, bytes can be useful as well.
 Although they seem so, these two functions are not so well-defined like `Clip`,`ToUint8`,`Resize` which you don't need to read docs to know what they do.
I don't mind removing `SelectComponent`, actually I would hope to let users get familiar with `MapData/MapImage` so they won't expect all simple functionalities built into tensorpack.